Sources/Frame.swift

```diff
@@ -36,3 +40,52 @@ struct Frame: Equatable, Hashable {
     }
 
 }
+
+private func computeViewport(from image: NSImage) -> NSRect {
+    let cgImage = image.cgImage(forProposedRect: nil, context: nil, hints: nil)!
+    let representation = NSBitmapImageRep(cgImage: cgImage)
+    let centerX = Int(image.size.width / 2)
+    let centerY = Int(image.size.height / 2)
+    let centerColor = representation.colorAt(x: centerX, y: centerY)
+    let x1 = computeHorizontalStartOf(centerColor, in: representation, startingFromX: centerX, y: centerY)
+    let x2 = computeHorizontalEndOf(centerColor, in: representation, startingFromX: centerX, y: centerY)
+    let y1 = computeVerticalStartOf(centerColor, in: representation, startingFromX: centerX, y: centerY)
+    let y2 = computeVerticalEndOf(centerColor, in: representation, startingFromX: centerX, y: centerY)
+    return NSRect(x: x1, y: y1, width: x2 - x1 + 1, height: y2 - y1 + 1)
```

Comment:
If it can't compute the viewport it will return a `NSRect` with height/width of 1, is that intended? Maybe have the `compute...()` functions return an optional and then throw an error if we can't calculate the bounds.

An alternative would be to return the image bounds as the viewport.

Reply:
Oh, one more thing, the expanded form of `x2 - x1 + 1` for the case we want comes to `(x + i - 1) - (x - i + 1) + 1` so a couple of the 1's get canceled out. Maybe moving the +/- 1 out of the `compute()` functions could help generalize it.